velox/functions/prestosql/window/tests/SimpleAggregatesTest.cpp

```diff
@@ -22,66 +22,98 @@ namespace facebook::velox::window::test {
+      std::string("sum(c2)"),
+      std::string("min(c2)"),
+      std::string("max(c2)"),
+      std::string("count(c2)"),
+      std::string("avg(c2)"),
+      std::string("sum(1)")};
+  std::vector<TestParam> params;
+
+  for (auto aggregateFunction : aggregateFunctions) {
+    for (auto frameClause : frameClauses) {
+      params.push_back({aggregateFunction, frameClause});
+    }
+  }
+  return params;
+}
+
 class SimpleAggregatesTest : public WindowTestBase {
  protected:
-  SimpleAggregatesTest(const std::string& function) : function_(function) {}
+  SimpleAggregatesTest(const TestParam& testParam)
+      : function_(testParam.aggregateFunction),
+        frameClause_(testParam.frameClause) {}
 
   RowVectorPtr makeBasicVectors(vector_size_t size) {
     return makeRowVector({
-        makeFlatVector<int64_t>(size, [](auto row) { return row % 10; }),
-        makeFlatVector<int64_t>(size, [](auto row) { return row % 7; }),
-        // k in k PRECEDING / FOLLOWING frame bounds, represented by c2 in
-        // certain testcases, should at least be 1.
+        makeFlatVector<int32_t>(size, [](auto row) { return row % 10; }),
+        // k in k PRECEDING / FOLLOWING frame bounds, represented by c1 and c2
+        // in certain testcases, should at least be 1.
+        makeFlatVector<int64_t>(size, [](auto row) { return row % 7 + 1; }),
         makeFlatVector<int64_t>(size, [](auto row) { return row + 1; }),
     });
   }
 
   RowVectorPtr makeSinglePartitionVector(vector_size_t size) {
     return makeRowVector({
-        makeFlatVector<int64_t>(size, [](auto /* row */) { return 1; }),
-        makeFlatVector<int64_t>(size, [](auto row) { return row % 50; }),
-        // k in k PRECEDING / FOLLOWING frame bounds, represented by c2 in
-        // certain testcases, should at least be 1.
+        makeFlatVector<int32_t>(size, [](auto /* row */) { return 1; }),
+        // k in k PRECEDING / FOLLOWING frame bounds, represented by c1 and c2
+        // in certain testcases, should at least be 1.
+        makeFlatVector<int64_t>(size, [](auto row) { return row % 50 + 1; }),
         makeFlatVector<int64_t>(size, [](auto row) { return row + 1; }),
     });
   }
 
   void testWindowFunction(
       const std::vector<RowVectorPtr>& vectors,
-      const std::vector<std::string>& overClauses,
-      const std::vector<std::string>& frameClauses = {""}) {
+      const std::vector<std::string>& overClauses) {
     WindowTestBase::testWindowFunction(
-        vectors, function_, overClauses, frameClauses);
+        vectors, function_, overClauses, frameClause_);
   }
 
   const std::string function_;
+  const std::string frameClause_;
+};
+
+class MultiFrameAggregatesTest : public SimpleAggregatesTest,
+                                 public testing::WithParamInterface<TestParam> {
+ public:
+  MultiFrameAggregatesTest() : SimpleAggregatesTest(GetParam()) {}
 };
 
 class MultiAggregatesTest : public SimpleAggregatesTest,
```

Comment:
Why do we need MultiFrameAggregatesTest and MultiAggregatesTest ? Both seem identical.

Reply:
Do we need to have 2 classes ? Can we just have 2 suite instantiations ?

Reply:
Thanks for the suggestion but 2 suite instantiations are not allowed.